plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -447,7 +451,15 @@ def get_metadata(self, metadata_files):
             metadata_files.download_metadata_files()
             self.downloader = None
             _logger.info(_('Generating metadata databases.'))
-            metadata_files.generate_dbs()
+
+            filelists_other_package_count = metadata_files.generate_dbs()
+            self.package_count.update(filelists_other_package_count)
+            if (self.package_count[filelists.METADATA_FILE_NAME] !=
+                    self.package_count[other.METADATA_FILE_NAME]):
+                reason = ('metadata is specified for different set of packages in filelists.xml'
```

Comment:
It is done in [one place where error code/message is defined](https://github.com/pulp/pulp_rpm/pull/1006/files#diff-140641dd82dde9606a1467743c1ebcdeR38).